pkg/validate/external.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	apierrors "github.com/operator-framework/api/pkg/validation/errors"
+)
+
+// For mocking in tests.
+var stderr io.Writer = os.Stderr
```

Comment:
This is unnecessary, you can redirect the actual `os.Stderr` variable in the test:

```go
var (
    origStderr io.Writer
    stderrBuf bytes.Buffer
)

BeforeEach(func() {
    origStderr = os.Stderr
    os.Stderr = stderrBuf
})

AfterEach(func() {
    stderrBuf.Reset()
    os.Stderr = origStderr
})
```

Works for all tests both inside and outside the package. Also doesn't require modification of the application code for testing.

Reply:
I left this as is for now.